File: example/functions/node_function/src/controllers/withParamsHandler.ts
import { RouteHandler } from "appwrouter";
import { AppwriteException } from "node-appwrite";

export const withParamsHandler: RouteHandler = async ({
  req,
  res,
  log,
  error,
  client,
}) => {
  try {
    const { studentId, id } = req.params;
    log("Hi Appwriters, this is a log message");
    log(JSON.stringify(req.params));
    return res.send(
      JSON.stringify({
        message: "Hello from Appwrouter!",
        data: {
          studentId,
          id,
        },
      }),
      201,
      {
        "content-type": "application/json",
      }
    );
  } catch (e) {
    error(` ${e.message}`);
    if (e instanceof AppwriteException) {
      return res.send(
        JSON.stringify({
          message: e.message,
        }),
        500,
        {
          "content-type": "application/json",
        }
      );
    } else {
      return res.send(
        JSON.stringify({
          message: "An error occurred",
        }),
        500,
        {
          "content-type": "application/json",
        }
      );
    }
  }
};

File: example/functions/dart_function/lib/controllers/with_params_handler.dart
import 'dart:convert';

import 'package:appwrouter/appwrouter.dart';
import 'package:dart_appwrite/dart_appwrite.dart';

Future<dynamic> withParamsHandler(HandleRequest handler) async {
  final HandleRequest(
    :req,
    :res,
    :error,
    :log,
  ) = handler;
  try {
    final params = req.params;
    log("Hi Appwriters, this is a log message");
    log(params.toString());
    return res.send(
        jsonEncode({
          "message": "Hello from Appwrouter!",
          "data": {
            "studentId": params["studentId"],
            "id": params["id"],
          }
        }),
        201,
        {
          "content-type": "application/json",
        });
  } catch (e) {
    error(e.toString());
    if (e is AppwriteException) {
      return res.send(
          jsonEncode({
            "message": e.message,
          }),
          500,
          {
            "content-type": "application/json",
          });
    } else {
      return res.send(
          jsonEncode({
            "message": "Internal Server Error",
          }),
          500,
          {
            "content-type": "application/json",
          });
    }
  }
}
